js/modules/k6/html/html.go

```diff
@@ -71,3 +99,438 @@ func (s Selection) Attr(name string, def ...goja.Value) goja.Value {
+}
+
+func (s Selection) Has(arg goja.Value) Selection {
+	return s.varargFnCall(arg, s.sel.Has, s.sel.HasSelection, s.sel.HasNodes)
+}
+
+func (s Selection) Map(v goja.Value) (result []string) {
+	gojaFn, isFn := goja.AssertFunction(v)
+	if isFn {
+		fn := func(idx int, sel *goquery.Selection) string {
+			if fnRes, fnErr := gojaFn(v, s.rt.ToValue(idx), s.rt.ToValue(sel)); fnErr == nil {
+				return fnRes.String()
+			} else {
+				return ""
+			}
+		}
+		return s.sel.Map(fn)
+	} else {
+		panic(s.rt.NewGoError(errors.New("Argument to map() must be a function")))
+		return nil
+	}
+}
+
+func (s Selection) Not(v goja.Value) Selection {
+	if gojaFn, isFn := goja.AssertFunction(v); isFn {
+		return Selection{s.rt, s.sel.NotFunction(s.buildMatcher(v, gojaFn))}
+	} else {
+		return s.varargFnCall(v, s.sel.Not, s.sel.NotSelection, s.sel.NotNodes)
+	}
+}
+
+func (s Selection) adjacent(unfiltered func() *goquery.Selection,
+	filtered func(string) *goquery.Selection,
+	def ...string) Selection {
+	if len(def) == 0 {
+		return Selection{s.rt, unfiltered()}
+	} else {
+		return Selection{s.rt, filtered(def[0])}
+	}
+}
+
+func (s Selection) Next(def ...string) Selection {
+	return s.adjacent(s.sel.Next, s.sel.NextFiltered, def...)
+}
+
+func (s Selection) NextAll(def ...string) Selection {
+	return s.adjacent(s.sel.NextAll, s.sel.NextAllFiltered, def...)
+}
+
+func (s Selection) Prev(def ...string) Selection {
+	return s.adjacent(s.sel.Prev, s.sel.PrevFiltered, def...)
+}
+
+func (s Selection) PrevAll(def ...string) Selection {
+	return s.adjacent(s.sel.PrevAll, s.sel.PrevAllFiltered, def...)
+}
+
+func (s Selection) Parent(def ...string) Selection {
+	return s.adjacent(s.sel.Parent, s.sel.ParentFiltered, def...)
+}
+
+func (s Selection) Parents(def ...string) Selection {
+	return s.adjacent(s.sel.Parents, s.sel.ParentsFiltered, def...)
+}
+
+func (s Selection) Siblings(def ...string) Selection {
+	return s.adjacent(s.sel.Siblings, s.sel.SiblingsFiltered, def...)
+}
+
+func (s Selection) adjacentUntil(until func(string) *goquery.Selection,
+	untilSelection func(*goquery.Selection) *goquery.Selection,
+	filteredUntil func(string, string) *goquery.Selection,
+	filteredUntilSelection func(string, *goquery.Selection) *goquery.Selection,
+	def ...goja.Value) Selection {
+	// empty selector to nextuntil and prevuntil acts like revAll and nextAll
+	// relies on goquery.compileMatcher returning a matcher which fails all matches when the selector being compiled is invalid
+	if len(def) == 0 {
+		return Selection{s.rt, until("")}
+	}
+
+	selector := def[0].Export()
+	if len(def) == 1 {
+		switch selector.(type) {
+		case string:
+			return Selection{s.rt, until(selector.(string))}
+
+		case Selection:
+			return Selection{s.rt, untilSelection(selector.(Selection).sel)}
+
+		case nil:
+			return Selection{s.rt, until("")}
+
+		default:
+			panic(s.rt.NewGoError(errors.New("Invalid argument. The selector must be a string or query object")))
```

Comment:
Query object isn't, jQuery object is what that was based on - eg https://api.jquery.com/add/#add-selection. Using 'jQuery object' felt wrong because it's really goquery... I'll try some variations.